opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -140,61 +141,80 @@ def collect(
         Atomically return a point for the current value of the metric and
         reset the aggregation value.
         """
-        if self._instrument_temporality is AggregationTemporality.DELTA:
 
-            with self._lock:
-                value = self._value
-                start_time_unix_nano = self._start_time_unix_nano
+        with self._lock:
```

Comment:
Discussed with @lzchen, could we add this picture as a comment in the source code somewhere? It's not specific to `Sum` so maybe at the module level

```
time --->

previous_point_cumulative:
|-----------------|

current_point_cumulative:
|-------------------------|

current_point_delta:
                  |-------|

Given you saved previous_point_cumulative from the previous collection:

- If you have current_point_delta (synchronous instrument), you can generate current_point_cumulative = previous_point_cumulative + current_point_delta
- If you have current_point_cumulative (async instrument), you can generate current_point_delta = current_point_cumulative - previous_point_cumulative
```

Reply:
Added the diagram and a more detailed explanation.